module/x/gravity/abci.go

```diff
@@ -32,8 +32,14 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) {
 }
 
 func createBatchTxs(ctx sdk.Context, k keeper.Keeper) {
-	// TODO: this needs some more work, is super naieve
-	if ctx.BlockHeight()%10 == 0 {
+	// TODO: this needs some more work, is super naive
```

Comment:
any context what this original TODO comment referred to? did it aim to make the period dynamic?

Reply:
I am not completely sure about the original comment.
But having a dynamic batch period would certainly provide us a better flexibility. It can be adjusted with average block time which may differ for each chains.

Another approach would be to use a probabilistic algorithm to time when batch needs to be recreated but it requires more thinking.